python/ql/lib/semmle/python/frameworks/Aiohttp.qll

```diff
@@ -621,14 +621,12 @@ module AiohttpWebModel {
     DataFlow::Node value;
 
     AiohttpResponseCookieSubscriptWrite() {
-      exists(SubscriptNode subscript |
-        // To give `this` a value, we need to choose between either LHS or RHS,
-        // and just go with the LHS
-        this.asCfgNode() = subscript
-      |
-        subscript.getObject() =
-          aiohttpResponseInstance().getMember("cookies").getAValueReachableFromSource().asCfgNode() and
-        value.asCfgNode() = subscript.(DefinitionNode).getValue() and
+      // `value` defines a subscript
+      value = aiohttpResponseInstance().getMember("cookies").getASubscript().asSink() and
+      // recover the `SubscriptNode` to get the index (and to define `this`)
+      // this part assumes a specific syntax for defining the subscript
+      exists(SubscriptNode subscript | value.asCfgNode() = subscript.(DefinitionNode).getValue() |
```

Comment:
I don't like having to drop down to the control-flow layer like this. We should add the appropriate classes to the API-graph and data-flow layers.

Reply:
I added subscript nodes, so now it is hidden in those. I could still also add a `DataFlow::SubscriptNode` to wrap `SubscriptNode` from the CFG. Then the hidden code would also be nice...